File: ontrack-extension-artifactory/src/main/java/net/nemerosa/ontrack/extension/artifactory/client/ArtifactoryClientFactoryImpl.java
package net.nemerosa.ontrack.extension.artifactory.client;

import net.nemerosa.ontrack.extension.artifactory.configuration.ArtifactoryConfiguration;
import net.nemerosa.ontrack.extension.support.client.ClientConnection;
import net.nemerosa.ontrack.extension.support.client.ClientFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

@Component
public class ArtifactoryClientFactoryImpl implements ArtifactoryClientFactory {

    private final ClientFactory clientFactory;

    @Autowired
    public ArtifactoryClientFactoryImpl(ClientFactory clientFactory) {
        this.clientFactory = clientFactory;
    }

    @Override
    public ArtifactoryClient getClient(ArtifactoryConfiguration configuration) {
        return new ArtifactoryClientImpl(
                clientFactory.getJsonClient(
                        new ClientConnection(
                                configuration.getUrl(),
                                configuration.getUser(),
                                configuration.getPassword()
                        )
                )
        );
    }
}

File: ontrack-extension-artifactory/src/main/java/net/nemerosa/ontrack/extension/artifactory/client/ArtifactoryClientFactoryImpl.kt
package net.nemerosa.ontrack.extension.artifactory.client

import net.nemerosa.ontrack.extension.artifactory.configuration.ArtifactoryConfiguration
import org.springframework.boot.web.client.RestTemplateBuilder
import org.springframework.stereotype.Component

@Component
class ArtifactoryClientFactoryImpl() :
    ArtifactoryClientFactory {

    override fun getClient(configuration: ArtifactoryConfiguration): ArtifactoryClient {
        val restTemplate = RestTemplateBuilder()
            .rootUri(configuration.url)
            .basicAuthentication(configuration.user, configuration.password)
            .build()
        return ArtifactoryClientImpl(restTemplate)
    }
}
